gr-uhd/lib/usrp_source_impl.cc

```diff
@@ -639,6 +640,12 @@ int usrp_source_impl::work(int noutput_items,
     case ::uhd::rx_metadata_t::ERROR_CODE_OVERFLOW: {
         _tag_now = true;
         ++_overflow_count;
+
+        const pmt::pmt_t val = pmt::PMT_NIL;
```

Comment:
These messages, if very frequent, could flood the message queue. They should still be limited, same as the log messages.

Reply:
ok! If that is the case maybe this could work:

```diff
diff --git a/gr-uhd/lib/usrp_source_impl.cc b/gr-uhd/lib/usrp_source_impl.cc
index fbd008df2..8a0f6b44b 100644
--- a/gr-uhd/lib/usrp_source_impl.cc
+++ b/gr-uhd/lib/usrp_source_impl.cc
@@ -641,11 +641,6 @@ int usrp_source_impl::work(int noutput_items,
         _tag_now = true;
         ++_overflow_count;
 
-        const pmt::pmt_t val = pmt::PMT_NIL;
-        pmt::pmt_t value = pmt::dict_add(pmt::make_dict(), EVENT_CODE_OVERFLOW, val);
-        pmt::pmt_t msg = pmt::cons(ASYNC_MSG_KEY, value);
-        message_port_pub(ASYNC_MSGS_PORT_KEY, msg);
-
         auto now = std::chrono::steady_clock::now();
         auto delta = now - _last_log;
         if (delta > _overflow_log_interval) {
@@ -654,6 +649,13 @@ int usrp_source_impl::work(int noutput_items,
                 std::chrono::duration_cast<std::chrono::milliseconds>(delta).count();
             d_logger->error(
                 "In the last {:d} ms, {:d} overflows occurred.", ms, _overflow_count);
+
+            pmt::pmt_t value = pmt::dict_add(pmt::make_dict(),
+                                             EVENT_CODE_OVERFLOW,
+                                             pmt::from_uint64(_overflow_count));
+            pmt::pmt_t msg = pmt::cons(ASYNC_MSG_KEY, value);
+            message_port_pub(ASYNC_MSGS_PORT_KEY, msg);
+
             _overflow_count = 0;
         }
         // ignore overflows and try work again
diff --git a/gr-uhd/lib/usrp_source_impl.h b/gr-uhd/lib/usrp_source_impl.h
index e883c9e3c..8cd986eb8 100644
--- a/gr-uhd/lib/usrp_source_impl.h
+++ b/gr-uhd/lib/usrp_source_impl.h
@@ -23,7 +23,7 @@ static const pmt::pmt_t RATE_KEY = pmt::string_to_symbol("rx_rate");
 static const pmt::pmt_t FREQ_KEY = pmt::string_to_symbol("rx_freq");
 static const pmt::pmt_t ASYNC_MSGS_PORT_KEY = pmt::string_to_symbol("async_msgs");
 static const pmt::pmt_t ASYNC_MSG_KEY = pmt::string_to_symbol("uhd_async_msg");
-static const pmt::pmt_t EVENT_CODE_OVERFLOW = pmt::string_to_symbol("overflow");
+static const pmt::pmt_t EVENT_CODE_OVERFLOW = pmt::string_to_symbol("overflows");
 
 namespace gr {
 namespace uhd {
```

Reply:
I think your solution is good -- can you please apply it? I'll unresolve this thread until you do.

Reply:
BTW, you can use the github suggestion feature to show changes inline.